dask_expr/_expr.py

```diff
@@ -2079,7 +2079,9 @@ def _fusion_pass(expr):
                         # of partitions, since broadcasting within
                         # a group is not allowed.
                         stack.append(dep)
-                    elif dep not in roots and dependencies[dep]:
+                    elif dependencies[dep] and dep._name not in [
+                        r._name for r in roots
+                    ]:
```

Comment:
It turns out `dep not in roots` doesn't actually work the way we want it to, and so we were previously missing some opportunities for fusion.

**Reproducer**:
```python
import dask_expr as dx
import pandas as pd

pdf = pd.DataFrame({"x": list(range(10)) * 10, "y": range(100), "z": 1})
df = dx.from_pandas(pdf)

assert df.expr not in [df[["x"]].expr]
```